pkg/kubelet/status/state/state_checkpoint.go

```diff
@@ -42,46 +42,47 @@ func NewStateCheckpoint(stateDir, checkpointName string) (State, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize checkpoint manager for pod allocation tracking: %v", err)
 	}
+
 	stateCheckpoint := &stateCheckpoint{
-		cache:             NewStateMemory(),
 		checkpointManager: checkpointManager,
 		checkpointName:    checkpointName,
 	}
 
-	if err := stateCheckpoint.restoreState(); err != nil {
+	praInfo, err := stateCheckpoint.restoreState()
+	if err != nil {
 		//lint:ignore ST1005 user-facing error message
 		return nil, fmt.Errorf("could not restore state from checkpoint: %v, please drain this node and delete pod allocation checkpoint file %q before restarting Kubelet", err, path.Join(stateDir, checkpointName))
 	}
+
+	stateCheckpoint.cache = NewStateMemory(praInfo.AllocationEntries)
+	klog.V(2).InfoS("State checkpoint: restored pod resource allocation state from checkpoint")
+
 	return stateCheckpoint, nil
 }
 
 // restores state from a checkpoint and creates it if it doesn't exist
-func (sc *stateCheckpoint) restoreState() error {
+func (sc *stateCheckpoint) restoreState() (*PodResourceAllocationInfo, error) {
 	sc.mux.Lock()
 	defer sc.mux.Unlock()
 	var err error
 
 	checkpoint, err := NewCheckpoint(nil)
 	if err != nil {
-		return fmt.Errorf("failed to create new checkpoint: %w", err)
+		return nil, fmt.Errorf("failed to create new checkpoint: %w", err)
 	}
```

Comment:
This is unnecessary. It should just be
```suggestion
checkpoint := &Checkpoint{}
```